File: server/mc/1.20.1-forge/src/main/java/calebxzhou/rdi/mc/server/tpa/TpaPlayer201.java
package calebxzhou.rdi.mc.server.tpa;

import calebxzhou.rdi.mc.common2.tpa.TpaPlayer;
import net.minecraft.network.chat.Component;
import net.minecraft.server.level.ServerPlayer;

import java.util.UUID;

public final class TpaPlayer201 implements TpaPlayer {
    private final ServerPlayer player;

    public TpaPlayer201(ServerPlayer player) {
        this.player = player;
    }

    public ServerPlayer unwrap() {
        return player;
    }

    @Override
    public UUID id() {
        return player.getUUID();
    }

    @Override
    public String name() {
        return player.getGameProfile().getName();
    }

    @Override
    public void sendMessage(String message) {
        player.sendSystemMessage(Component.literal(message));
    }
}

File: server/mc/1.20.1-forge/src/main/kotlin/calebxzhou/rdi/mc/server/tpa/TpaPlayer201.kt
package calebxzhou.rdi.mc.server.tpa

import calebxzhou.rdi.mc.rcmd.tpa.TpaPlayer
import net.minecraft.network.chat.Component
import net.minecraft.server.level.ServerPlayer
import java.util.UUID

class TpaPlayer201(private val player: ServerPlayer) : TpaPlayer {

    fun unwrap(): ServerPlayer {
        return player
    }

    public override fun id(): UUID {
        return player.getUUID()
    }

    public override fun name(): String {
        return player.gameProfile.name
    }

    public override fun sendMessage(message: String) {
        player.sendSystemMessage(Component.literal(message))
    }
}
